proto/viam/app/mltraining/v1/ml_training.proto

```diff
@@ -19,6 +19,7 @@ enum ModelType {
   MODEL_TYPE_UNSPECIFIED = 0;
   MODEL_TYPE_SINGLE_LABEL_CLASSIFICATION = 1;
   MODEL_TYPE_MULTI_LABEL_CLASSIFICATION = 2;
+  MODEL_TYPE_OBJECT_DETECTION = 3;
```

Comment:
Note: made this OBJECT_DETECTION based on what it's commonly called, vs needing to prepend IMAGE.